integration/maptest/map_test.go

```diff
@@ -15,409 +15,54 @@
-
-				if got, want := getResp.MapLeafInclusion[0].GetLeaf().GetLeafValue(), batch.LeafValue; !bytes.Equal(got, want) {
-					t.Errorf("GetLeaves(rev: %v).LeafValue: %s, want %s", batch.revision, got, want)
-					continue
-				}
 
-				if err := verifyGetMapLeavesResponse(getResp, indexes, batch.revision,
-					pubKey, hasher, tree.TreeId); err != nil {
-					t.Errorf("%v: verifyGetMapLeavesResponse(rev %v): %v", tc.desc, batch.revision, err)
-					continue
-				}
-			}
-		}
-	}
+	os.Exit(m.Run())
 }
 
-func TestInclusion(t *testing.T) {
+func TestMapIntegration(t *testing.T) {
 	ctx := context.Background()
-	for _, tc := range []struct {
-		desc         string
-		HashStrategy []trillian.HashStrategy
-		leaves       []*trillian.MapLeaf
-	}{
-		{
-			desc:         "single",
-			HashStrategy: []trillian.HashStrategy{trillian.HashStrategy_TEST_MAP_HASHER, trillian.HashStrategy_CONIKS_SHA512_256},
-			leaves: []*trillian.MapLeaf{
-				{Index: h2b("0000000000000000000000000000000000000000000000000000000000000000"), LeafValue: []byte("A")},
-			},
-		},
-		{
-			desc:         "multi",
-			HashStrategy: []trillian.HashStrategy{trillian.HashStrategy_TEST_MAP_HASHER, trillian.HashStrategy_CONIKS_SHA512_256},
-			leaves: []*trillian.MapLeaf{
-				{Index: h2b("0000000000000000000000000000000000000000000000000000000000000000"), LeafValue: []byte("A")},
-				{Index: h2b("0000000000000000000000000000000000000000000000000000000000000001"), LeafValue: []byte("B")},
-				{Index: h2b("0000000000000000000000000000000000000000000000000000000000000002"), LeafValue: []byte("C")},
-				{Index: h2b("0000000000000000000000000000000000000000000000000000000000000003"), LeafValue: nil},
-			},
-		},
-		{
-			desc:         "across subtrees",
-			HashStrategy: []trillian.HashStrategy{trillian.HashStrategy_TEST_MAP_HASHER, trillian.HashStrategy_CONIKS_SHA512_256},
-			leaves: []*trillian.MapLeaf{
-				{Index: h2b("0000000000000180000000000000000000000000000000000000000000000000"), LeafValue: []byte("Z")},
-			},
-		},
-	} {
-		for _, hashStrategy := range tc.HashStrategy {
-			tree, hasher, err := newTreeWithHasher(ctx, env, hashStrategy)
-			if err != nil {
-				t.Errorf("%v: newTreeWithHasher(%v): %v", tc.desc, hashStrategy, err)
-				continue
-			}
-			pubKey, err := der.UnmarshalPublicKey(tree.GetPublicKey().GetDer())
-			if err != nil {
-				t.Errorf("%v: UnmarshalPublicKey(%v): %v", tc.desc, hashStrategy, err)
-				continue
-			}
-
-			if _, err := env.MapClient.SetLeaves(ctx, &trillian.SetMapLeavesRequest{
-				MapId:  tree.TreeId,
-				Leaves: tc.leaves,
-				Metadata: testonly.MustMarshalAny(t, &ctmapperpb.MapperMetadata{
-					HighestFullyCompletedSeq: 0xcafe,
-				}),
-			}); err != nil {
-				t.Errorf("%v: SetLeaves(): %v", tc.desc, err)
-				continue
-			}
-
-			indexes := [][]byte{}
-			for _, l := range tc.leaves {
-				indexes = append(indexes, l.Index)
+	for _, test := range AllTests {
+		t.Run(test.Name, func(t *testing.T) {
```

Comment:
What do you think about passing the `testing.T` object to the subtests? 
That way we preserve the line number where things went wrong

Reply:
I think this is the correct way to use subtests.

The testing.Run implementation does create a new testing.T to pass into the sub-test.

If you pass in the parent one (by declaring the sub-test as func(*testing.T) omitting the arg name) then errors from the subtest get mis-attributed